packages/mainmenu-extension/src/index.ts

```diff
@@ -485,6 +493,17 @@ function createTabsMenu(app: JupyterLab, menu: TabsMenu): void {
     return { command: commandID };
   };
 
+  let previousId = '';
+
+  // Command to toggle between the current
+  // tab and the previously used tab.
+  if (!commands.hasCommand(CommandIDs.activatePreviouslyUsedTab)) {
+    commands.addCommand(CommandIDs.activatePreviouslyUsedTab, {
+      label: 'Activate Previously Used Tab',
+      execute: () => previousId && app.commands.execute(`tabmenu:activate-${previousId}`)
```

Comment:
We can also grey out the menu item when there is no previously used tab by adding an `isEnabled` property here. Something like
```typescript
isEnabled: () => !!previousId
```

Reply:
Regarding `isEnabled`, it would be nice to disable the command if the user manually 'x'-s out the previously used tab. I am wondering if there is an event that fires in such a scenario. I would then check if the `id` of that widget matches `previousId`  and null out the latter.

Reply:
That's a nice idea. We could probably do it with a `disposed` signal, as you suggest (there is no problem with more than one connection to a given signal). I wonder if it would be easier to scan through the widgets in the main area to see if they are still there. Something like
```typescript
isEnabled: () => previousId && find(app.shell.widgets('main), widget => widget.id === previousId )
```

Reply:
Alternatively, since `populateTabs` is called every time that the layout is modified (including when tabs are closed), the `find` function I described could also do the job of clearing out the `previousId` string there. So it could be
```typescript
    const populateTabs = () => {
      menu.removeGroup(tabGroup);
      tabGroup.length = 0;
      each(app.shell.widgets('main'), widget => {
        tabGroup.push(createMenuItem(widget));
      });
      if (!find(app.shell.widgets('main'), widget => widget.id === previousId) {
        previousId = '';
      }
      menu.addGroup(tabGroup, 1);
    };
```

Reply:
I like the alternative approach.  One wouldn't necessarily expect people to have a ton of tabs open at one time, but are we o.k. with the possible time hit using `find` introduces?

Reply:
I think it should be okay. In fact, we are *already* iterating over the main area widgets, so that check can even go into the `each` function above.

Reply:
```typescript
   const populateTabs = () => {
      menu.removeGroup(tabGroup);
      tabGroup.length = 0;
      let isPreviouslyUsedTabAttached = false;
      each(app.shell.widgets('main'), widget => {
        if (widget.id === previousId) {
          isPreviouslyUsedTabAttached = true;
        }
        tabGroup.push(createMenuItem(widget));
      });
      menu.addGroup(tabGroup, 1);
      previousId = isPreviouslyUsedTabAttached ? previousId : '';
    };
   ```

Reply:
Yep, that would work nicely, I think. And it only involves iterating over the widgets once.